tasks/sast-go.yaml

```diff
@@ -0,0 +1,39 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: gosec-check
+spec:
+  workspaces:
+    - name: workspace
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  steps:
+    - name: gosec-check
+      image: quay.io/redhat-appstudio/hacbs-test:stable
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      script: |
+        /usr/local/go/bin/gosec -no-fail -fmt=sarif -out=gosec_output.json $(workspaces.workspace.path)/... | tee gosec_output.txt
+        GOSEC_EXIT_CODE=$?
+
+        # Test if any package was found
+        # Even with -no-fail, gosec uses exit code 1 for several states,
+        # including when there are no packages found.
+        grep "No packages found$" gosec_output.txt && echo "Exit 0 because no packages found" && exit 0
+        exit $GOSEC_EXIT_CODE
```

Comment:
it should never end with different exit than 0

Reply:
That's the problem I was solving here, perhaps it's a bit clunky. However, as the comment says, if we run the task for some repo that doesn't contain any Go code (the minimal reproducible case is by running `./hack/test-build.sh`), then even with `-no-fail`, gosec will exit with code 1. It will however print `No packages found` to stdout, which is the only indication (afaik, I might be wrong) for us that we should not fail the task as a whole. So in the case it finds no packages, line 23 will `exit 0` and if there is any package, we'll exit with whatever gosec decided to exit with. That would _usually_ be 0 but also not always. I was reading gosec's code and unfortunately I think this is what the developers left us with.

However now after typing my comment, I have a feeling that you might have meant something else, like that the task itself shouldn't exit with anything else than 0? If that's the case, no problem with reworking this, however I am still a bit confused about your other comment about `bash -ex` and how to handle non-Go repos.